src/Plugin/views/access/EntityOperation.php

```diff
@@ -33,40 +27,27 @@ class EntityOperation extends AccessPluginBase {
   protected $usesOptions = TRUE;
 
   /**
-   * The entity manager.
+   * The entity type manager.
    *
-   * @var \Drupal\Core\Entity\EntityManagerInterface
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
    */
-  protected $entityManager;
+  protected $entityTypeManager;
 
   /**
-   * Constructs a EntityOperation object.
+   * The entity type repository.
    *
-   * @param array $configuration
-   *   A configuration array containing information about the plugin instance.
-   * @param string $plugin_id
-   *   The plugin_id for the plugin instance.
-   * @param mixed $plugin_definition
-   *   The plugin implementation definition.
-   * @param \Drupal\Core\Entity\EntityManagerInterface $entity_manager
-   *   The entity manager.
+   * @var \Drupal\Core\Entity\EntityTypeRepositoryInterface
    */
-  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityManagerInterface $entity_manager) {
-    parent::__construct($configuration, $plugin_id, $plugin_definition);
-
-    $this->entityManager = $entity_manager;
-  }
+  protected $entityTypeRepository;
 
   /**
    * {@inheritdoc}
    */
   public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
-    return new static(
-      $configuration,
-      $plugin_id,
-      $plugin_definition,
-      $container->get('entity.manager')
-    );
+    $instance = new static($configuration, $plugin_id, $plugin_definition);
```

Comment:
This is a bit of a divergence from conventions. I'm not ready to consider it for long established code. Constructor should return.

Reply:
It's a new pattern to use to have less issues with backward compatibility in future, because contrib code has no control on extended parent class constructor (it's started to be adopted in latest releases of Drupal core)
Ref https://www.previousnext.com.au/blog/safely-extending-drupal-8-plugin-classes-without-fear-of-constructor-changes

Reply:
Thanks! Merged